pkg/gitreceive/git/sha.go

```diff
@@ -0,0 +1,37 @@
+package git
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const (
+	// this constant represents the length of a shortened git sha - 8 characters long
+	shortShaIdx = 8
+	fullShaLen  = 40
+)
+
+var shaRegex = regexp.MustCompile(`^[\da-f]{40}$`)
+
+type ErrInvalidGitSha struct {
+	sha string
+}
+
+func (e ErrInvalidGitSha) Error() string {
+	return fmt.Sprintf("git sha %s was invalid", e.sha)
+}
+
+type SHA struct {
+	full  string
+	short string
+}
+
+func NewSha(rawSha string) (*SHA, error) {
```

Comment:
Ah, I remember what my knowledge is coming from. In deisctl we had a function called `.FleetClient()` in package `client/fleet`. go vet didn't like it due to stuttering and so it preferred `.New` instead. That function was created before we introduced go vet though and we never got around to fixing it.